lazyflow/slot.py

```diff
@@ -483,143 +490,71 @@ def _handleUpstreamUnready(self, slot):
-                                   not self.meta._ready)
-                    self.meta = self.upstream_slot.meta.copy()
-    
-                    # the slot with more sub-slots determines
-                    # the number of subslots
-                    if len(self) < len(upstream_slot):
-                        self.resize(len(upstream_slot))
-                    elif len(self) > len(upstream_slot):
-                        upstream_slot.resize(len(self))
-    
-                    upstream_slot.downstream_slots.append(self)
-                    for i in range(len(self.upstream_slot)):
-                        p = self.upstream_slot[i]
-                        self[i].connect(p)
-    
-                    # call slot type connect function
-                    self.stype.connect(upstream_slot)
-    
-                    if self.level > 0 or self.stype.isConfigured():
-                        self._changed()
-    
-                    # call connect callbacks
-                    self._sig_connect(self)
-    
-                    # Notify readiness after upstream_slot is updated
-                    if notifyReady:
-                        self._sig_ready(self)
-    
-                elif upstream_slot.level < self.level:
-                    self.upstream_slot = upstream_slot
-                    notifyReady = (self.upstream_slot.meta._ready and not
-                                   self.meta._ready)
-                    self.meta = self.upstream_slot.meta.copy()
-                    for i, slot in enumerate(self._subSlots):
-                        slot.connect(upstream_slot)
-    
-                    if notifyReady:
-                        self._sig_ready(self)
-    
-                    self._changed()
-                    # call connect callbacks
-                    self._sig_connect(self)
-    
-                elif upstream_slot.level > self.level:
-                    msg = str("Can't connect slots:"
-                           " {}.{}.level={}, but"
-                           " {}.{}.level={}"
-                           " (Implicit OpearatorWrapper creation"
-                           " is no longer supported.)").format(
-                               self.getRealOperator().name,
-                               self.name, self.level,
-                               upstream_slot.getRealOperator().name,
-                               upstream_slot.name, upstream_slot.level)
-                    raise RuntimeError(msg)
-    
-                # propagate value changed signals from inner to outer
-                # operators.
-                if self._type == upstream_slot._type == "output":
-                    upstream_slot.notifyValueChanged(self._sig_value_changed)
+        if self.upstream_slot is upstream_slot and upstream_slot.level == self.level:
+            return
 
-        except:
-            try:
-                raise
-            finally:
-                try:
```

Comment:
Have you figured out what was going on here? I was scared of touching that part...